File: kotlin/aes(GUI)/src/main/kotlin/com/example/demo/AesFunctions/RoundObject.kt
package AesFunctions

class RoundObject(val isPreRound: Boolean = false) {
    var preRoundState: String = ""
    var subBytesState: String = ""
    var shiftRowsState: String = ""
    var mixColumnsState: String = ""
    var addRoundKeyState: String = ""
}

File: matlab/aes(GUI)/RoundObject.m
classdef RoundObject
    %RoundObject Object that holds the state values for each round
    
    
    properties
        preRoundState
        subBytesState
        shiftRowsState
        mixColumnsState
        addRoundKeyState
        isPreRound
    end
    
    methods
        function obj = RoundObject()
            obj.preRoundState = '';
            obj.subBytesState = '';
            obj.shiftRowsState = '';
            obj.mixColumnsState = '';
            obj.addRoundKeyState = '';
            obj.isPreRound = false;
        end
    end
    
end

